pubsub/google/cloud/pubsub_v1/subscriber/_consumer.py

```diff
@@ -392,3 +425,27 @@ def stop_consuming(self):
         """
         thread = self._stop_no_join()
         thread.join()
+
+
+def _pausable_iterator(iterator, can_continue):
```

Comment:
I toyed with a [`Pausable`][1] class but I think it makes more sense to store the `threading.Event` for `can_continue` / `can_consume` on the `Consumer` instance.

[1]: https://gist.github.com/dhermes/7aa696229833ead1a414f87697d82b46

Reply:
I'm okay with less classes.

Reply:
Yeah, objects seem super cumbersome when we have so many related but independent workers.